### ISSUE TITLE
Single Sign On Like PhpMyAdmin

### ISSUE THREAD

**Issue by ghost** [osTicket/osTicket-1.7#431]
PhpMyAdmin has the following feature:  http://wiki.phpmyadmin.net/pma/auth_types#signon

So if you set the following in your application login code:

```
$_SESSION['PMA_single_signon_user'] = $_POST['user'];
$_SESSION['PMA_single_signon_password'] = $_POST['password'];
```

and then visit PhpMyAdmin it wil log you in with those credentials. Basically PhpMyAdmin when authtype is set to 'signon' will initiate a session using the current set session and check to see if those session values are present, if they are it will perform a login with them.

It's as simple as having /login.php check for say $_SESSION['OST_single_signon_user']  and $_SESSION['OST_single_signon_user']  and performing a login with them. CSRF check is not needed as we know that the user has already been authenticated otherwise those $_SESSION vars would not exist.

This looks like something OST could implement as well since having SSO capabilities for support systems would allow multiple systems from OST, billing, other systems to login with a single login (same username and password for all systems of course). They don't have to share sessions necessarily as once authenticated it can utilize it's own session. This would be an optional config item.

I would do this myself and make a pull request, but what I have observed from other authentication posts OST devs are very cautious of modifications to authentication code in OST due to potential security issues. However, this should pose no threat as far as I can see as the CSRF check would be pointless as the developer would be the one saying yes this user is authenticated by setting those session vars in thier app, and the login credentials would be the same as if the login form was submitted by itself just without CSRF check.

If this is something that would be ok with you guys would it be alright if I coded it in and made a pull request? I don't want to do it unless it will make it into OST so that others can benefit from it.

If yes, are there any suggestions or concerns you think I should consider?
If not, would this be possible for OST devs to add in?

**Comment by rajagopalan** [comment]
Yes i would be ok with your suggestion and i welcome your suggestion.

**Comment by sabas** [comment]
SSO for staff session would require a modified version of login() from class.staff.php and login.php to be modified to call this modified function when sso is needed.. (Don't know about security issues :) )
